code/modules/power/lighting.dm

```diff
@@ -359,10 +353,15 @@
 
 /obj/machinery/light/update_overlays()
 	. = ..()
+	SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays)
 	if(on && status == LIGHT_OK)
-		glowybit.alpha = clamp(light_power*250, 30, 200)
-	else
-		glowybit.alpha = 0
+		var/area/A = get_area(src)
+		if(emergency_mode || (A?.fire))
+			SSvis_overlays.add_vis_overlay(src, overlayicon, "[base_state]_emergency", layer, plane, dir)
+		else if (nightshift_enabled)
+			SSvis_overlays.add_vis_overlay(src, overlayicon, "[base_state]_nightshift", layer, plane, dir)
+		else
+			SSvis_overlays.add_vis_overlay(src, overlayicon, base_state, layer, plane, dir)
```

Comment:
This seems to be new behavior, are you just adding new variants? where did the alpha dimming behavior go?

Reply:
This is adding new variants, to make it so the SSvis has only 3 overlays instantiated for all light fixtures, instead of having 1 unique overlay for each one

Reply:
And alpha doesnt need to be set to 0 as the overlay will be cut when the light fixture is turned off/broken

Reply:
I suppose the only visual difference is when you have a dim lights, then their shiny overlay will seem like the 100% alpha, instead of the 70% or so.